src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -199,14 +237,21 @@ def get_filename(file_type, file_prefix, matrix_combo):
     """
     file_type_prefix = ""
     file_ext = ""
+    file_name_prefix = file_key
     if file_type == str(OutputTypes.CONDA):
         file_ext = ".yaml"
-    if file_type == str(OutputTypes.REQUIREMENTS):
+    elif file_type == str(OutputTypes.REQUIREMENTS):
         file_ext = ".txt"
         file_type_prefix = "requirements"
+    elif file_type == str(OutputTypes.PYPROJECT):
+        file_ext = ".toml"
+        # Unlike for files like requirements.txt or conda environment YAML files, which
+        # may be named with additional prefixes (e.g. all_cuda_*) pyproject.toml files
+        # need to have that exact name and are never prefixed.
+        file_name_prefix = "pyproject"
     suffix = "_".join([f"{k}-{v}" for k, v in matrix_combo.items()])
     filename = "_".join(
-        x for x in [file_type_prefix, file_prefix, suffix] if x
+        x for x in [file_type_prefix, file_name_prefix, suffix] if x
     ).replace(".", "")
```

Comment:
This can be simplified to:
```python
filename = "_".join(filter(None, (file_type_prefix, file_name_prefix, suffix))).replace(".", "")
```

Reply:
Nice use of filter, I never remember the None option.